addons/purchase_stock/report/report_stock_forecasted.py

```diff
@@ -22,6 +22,7 @@ def _compute_draft_quantity_count(self, product_template_ids, product_variant_id
             in_sum = sum(quantities)
         res['draft_purchase_qty'] = in_sum
         res['draft_purchase_orders'] = po_lines.mapped("order_id").sorted(key=lambda po: po.name)
+        res['draft_purchase_orders_matched'] = self.env.context.get('purchase_line_to_match_id') in [purchase_order.id for purchase_order in po_lines]
```

Comment:
`[purchase_order.id for purchase_order in po_lines]` is ambigus (purchase_order ?). you can simplify it by `in po_lines.ids`